util/compression.h

```diff
@@ -158,13 +158,13 @@ struct CompressionDict {
 #if ZSTD_VERSION_NUMBER >= 700
   ZSTD_CDict* zstd_cdict_ = nullptr;
 #endif  // ZSTD_VERSION_NUMBER >= 700
-  Slice dict_;
+  std::string dict_;
 
  public:
 #if ZSTD_VERSION_NUMBER >= 700
-  CompressionDict(Slice dict, CompressionType type, int level) {
+  CompressionDict(std::string dict, CompressionType type, int level) {
 #else   // ZSTD_VERSION_NUMBER >= 700
-  CompressionDict(Slice dict, CompressionType /*type*/, int /*level*/) {
+  CompressionDict(std::string dict, CompressionType /*type*/, int /*level*/) {
```

Comment:
Hm I usually pass-by-value when possible. It gives the caller the decision whether to make a copy or do a move. pass-by-rvalue-reference can prevent the caller from calling the function in a way that invokes copy constructor, but I do not know we need to prevent that here.

Reply:
SGTM.